chain/chain/src/store.rs

```diff
@@ -142,6 +142,20 @@ pub trait ChainStoreAccess {
     fn get_block_header(&mut self, h: &CryptoHash) -> Result<&BlockHeader, Error>;
     /// Returns hash of the block on the main chain for given height.
     fn get_block_hash_by_height(&mut self, height: BlockHeight) -> Result<CryptoHash, Error>;
+    /// Returns hash of the first available block after genesis.
+    fn get_earliest_block_hash(&mut self) -> Result<Option<CryptoHash>, Error> {
+        let head_header_height = self.head_header()?.height();
+        let tail = self.tail()?;
+        if let Ok(block_hash) = self.get_block_hash_by_height(tail) {
+            return Ok(Some(block_hash.clone()));
+        }
+        for height in tail + 1..=head_header_height {
```

Comment:
Let's say the node misses some blocks, how expensive is this loop in the worst case scenario long-term?

Reply:
In the normal case, where a node is mostly online and don't miss blocks (we optimize our RPC nodes for this), this should not cause problems.

@Kouprin What do you think?

Reply:
if a node misses some blocks it will either sync the blocks or sync to the network through state sync, in which case tail will be reset. The only case where this can get be expensive is when a lot of heights are skipped. However, since doomslug has liveness, this should not be an issue.

Reply:
> Let's say the node misses some blocks

@evgenykuzyakov We are getting Tail of Canonical Chain which is five epochs behind the Head in average. Not sure if your question is relevant.

@frol I agreed with @bowenwang1996, it should be fine in normal circumstances. In case of skipping lots of Blocks in Canonical Chain for a long time we will face way more unpleasant consequences like crashing the node.